rfcs/debuggability_metrics.md

```diff
@@ -0,0 +1,126 @@
+   1. Time when failure is first signaled inside Jenkins job
+   2. Time when Jenkins sends an e-mail reporting the failure to the user (or a bot posts a failure message)
+   3. Time when user takes action on the notification and starts working on their machine (modifying code, or starting a build)
+2. Log-finding time: This step would have two parts. We expect this to be a small portion of the total time consumed, but it is still valuable to reduce it where possible.
+   1. Time it takes from notification to finding the correct log
+   2. Time it takes (after finding the correct log) to parse enough information in the log to take action locally
+3. Artifact-finding time: We expect this to be a small portion of the total time consumed, but it is still valuable to reduce it where possible.
+
+The first is a binary option - we can either send the e-mail at the end of the build, or upon the first failure. But measuring that difference could allow us to decide if the earlier messaging time is worthwhile. The second two options are going to have to be user-reported metrics.
+
+Metric | Description | Measured By
+-- | -- | --
+Failure first signaled | Time to see first failure in Jenkins | Automated script
+Failure e-mail sent | How long it takes for an e-mail/message to go out after a failure is found | Automated script
+Log-finding | How long it takes & how many clicks to find the appropriate log from a failed Jenkins run main page | Polling, customer reports
+Log-parsing | How long it takes to find the actual error message in the appropriate log | Polling, customer reports
+Artifact-finding | How long it takes & how many clicks to find artifacts from a failed Jenkins run main page | Polling, customer reports
+
+## Steps to Achieve Goal
+
+Even if a given objective is swift to complete, having too many steps (often in tedious clicks), can be a frustrating developer experience. Thus, we will measure the amount of steps a given goal took prior to changes and after changes. This metric has the advantage of not relying on polling, either, as we can independently measure this.
+
+## Polling
+
+While potentially less reliable than absolute metrics such as measuring time, polling customers for their preferences and pain points is still valuable. We would reach out to the sig-ux in order to develop questions. Measuring user preferences is not a primary skill set of our team and we would want to lean on experts to ensure an effective solution for our customers.
+
+In addition to broad polls, we will include deeper individual customer reports where we work with a single developer to examine their experience directly. For this step, we would work with a small sample of specific developers to get a detailed view of their debugging experience before and after changes. While this will take more time, it will provide more reliable results and allow us to guarantee accuracy in our polling.
```

Comment:
The time concern is already mentioned here but I just want to highlight it because it may affect other work by slowing  the time it takes to complete while they're manually filling out this metric info. We need to make sure this process is as efficient as possible and should automate as much of it as possible.

Basically, anything manually done should be short, in order to keep our workflow efficient.